### PR TITLE
DDP-5666: Implement a new vertical progress indicator

### PR DESCRIPTION
**Pull request by alextanlan**
Implement a new vertical progress indicator and configure it for dealing with Family History activity

---
This change is [Reviewable](https://reviewable.io/reviews/broadinstitute/ddp-angular/473)

### CODE REVIEW

**Comment by MocanaAtBroad** [comment]
New changes appear to break the Osteo Count Me In Prequalifier. The form is not displayed at all.
1) Run Osteo locally, pointing to back end in dev.
2) Clear all local caches.
3) Click on the Count Me In button in front/welcome page:
Screen Shot 2021-03-15 at 4 18 59 PM

**Comment by MocanaAtBroad** [comment]
Also, I see that we did some refactoring, moving activity-redesigned to a new folder.
I think this makes sense if we are going to split the typescript from the html, but now looking at diffs does not work; cannot really see what code changes are new, so a lot harder to review.
Would it be too difficult bring back components/activityForm/activity-redesigned.component.ts and make your changes there? Would help review. We can refactor files to new location afterwards.

**Comment by MocanaAtBroad** [comment]
Would be helpful if you could add a few comments either to code or itself or PR (whichever you think is more appropriate) that  can provide a trail of how you approached the changes.

**Comment by alextanlan** [comment]
>Also, I see that we did some refactoring, moving activity-redesigned to a new folder.
I think this makes sense if we are going to split the typescript from the html, but now looking at diffs does not work; cannot really see what code changes are new, so a lot harder to review.
Would it be too difficult bring back components/activityForm/activity-redesigned.component.ts and make your changes there? Would help review. We can refactor files to new location afterwards.

Sure. I returned back activity-redesigned.component.ts to /activityForm folder as to have more clear code-review.

**Comment by alextanlan** [comment]
> New changes appear to break the Osteo Count Me In Prequalifier. The form is not displayed at all.

Thank you for noticing!
A closing tag was wrongly placed. Fixed for now at f21ce72e28f7ed80e7e34ce86c05feaeecc3a6be.

File: ddp-workspace/projects/ddp-osteo/src/styles/activity.scss [hunk 1]
```diff
@@ -80,6 +92,93 @@
     margin: 3rem 0;
 }
 
+.vertical {
+    &.activity-steps {
+        grid-gap: 0.5rem;
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-osteo/src/styles/activity.scss:97]: Customized styles for the vertical progress indicator.

File: ddp-workspace/projects/ddp-osteo/src/styles/activity.scss [hunk 2]
```diff
@@ -80,6 +92,93 @@
...
+            justify-content: center;
+        }
+
+        &__text {
+            margin-left: 0.5rem;
+        }
+
+        @media only screen and (max-width: 480px) {
+            &__text {
+                display: none;
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-osteo/src/styles/activity.scss:176]: Hid steps labels for small screens.

File: ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-redesigned.component.html [hunk 3]
```diff
@@ -0,0 +1,208 @@
+<main class="main main_activity"
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-redesigned.component.html:1]: - Extracted the html template into a separate file.
- Refactored in order to have more readable template and decrease code lines.
- Used (hidden/non-displayed) templates as much as possible (^ refactoring) and placed them below the main template.

File: ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-redesigned.component.html [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+            <div class="content content_tight">
+                <h1 class="activity-header" [innerHTML]="model.title"></h1>
+            </div>
+        </section>
+    </ng-container>
+
+    <!-- article content -->
+    <section *ngIf="isLoaded; else spinner" class="section article-content">
+        <ng-container *ngIf="isVerticalProgress && isStepped && showStepper; then verticalProgressIndicator;">
+        </ng-container>
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-redesigned.component.html:31]: Using the vertical progress indicator there (as an external block on the left of a main article content)

File: ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-redesigned.component.html [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+                    (visibilityChanged)="updateVisibility($event)"
+                    (embeddedComponentsValidationStatus)="updateEmbeddedComponentValidationStatus(0, $event)"
+                    (embeddedComponentBusy)="embeddedComponentBusy$[0].next($event)">
+                </ddp-activity-section>
+            </ng-container>
+
+            <ng-container *ngIf="isStepped && showStepper">
+                <ng-container *ngIf="!isVerticalProgress; then horizontalProgressIndicator">
+                </ng-container>
+            </ng-container>
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-redesigned.component.html:56]: Using a "default" horizontal progress indicator there (inside or a main article content).

File: ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-redesigned.component.html [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+                    <span class="last-updated">{{model.lastUpdatedText}} </span>
+                </ng-container>
+
+                <div class="activity-buttons"
+                     [ngClass]="{'activity-buttons_mobile': (!isStepped || isLastStep) && isAgree() && isLoaded && !isReadonly()}">
+                    <ng-container *ngIf="isLoaded && isStepped; then navigationButtons"></ng-container>
+                    <ng-container *ngIf="(!isStepped || isLastStep) && isLoaded; then lastStepButtons"></ng-container>
+                    <ng-container
+                        *ngIf="(!isStepped || isLastStep) && isAgree() && isLoaded && !isReadonly(); then agreementButtons">
+                    </ng-container>
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-redesigned.component.html:98]: Simplified activity buttons block, using templates

File: ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-redesigned.component.html [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+                </div>
+                <div *ngIf="communicationErrorOccurred" class="ErrorMessage">
+                    <span translate>SDK.CommunicationError</span>
+                </div>
+            </ng-container>
+        </div>
+    </section>
+</main>
+
+<!--------------------------Used templates------------------------------>
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-redesigned.component.html:112]: Extracted templates for using in the template above. 
The templates are not  visible on the real page.

File: ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-redesigned.component.html [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+<ng-template #spinner>
+    <section class="section section-spinner">
+        <mat-spinner></mat-spinner>
+    </section>
+</ng-template>
+
+<ng-template #stepsWithCircle>
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-redesigned.component.html:119]: Used this (already implemented) template for both types of progress indicators: horizontal / vertical.
Custom styles are provided in a proper activity (in ddp-osteo for this particular PR)

File: ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/baseActivity.component.ts [hunk 9]
```diff
@@ -107,6 +108,7 @@ export abstract class BaseActivityComponent implements OnChanges, OnDestroy {
                     this.activityCode.emit(this.model.activityCode);
                     this.initSteps();
                 }
+                this.isLoaded$.next();
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/baseActivity.component.ts]: Emit an event when a model is loaded.
It is used in ddp-sdk/../activity-redesigned.component.ts as to check whether the activity should use vertical progress indicator (_isVerticalProgress_ flag).

File: ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-redesigned.component.html [hunk 10]
```diff
@@ -0,0 +1,208 @@
...
+    <div class="activity-steps" [ngClass]="{'vertical': isVerticalProgress}">
+        <ng-container *ngFor="let section of model.sections; index as i; last as isLastStep">
+            <ng-container *ngIf="section.visible">
+                <div class="activity-step" cdkMonitorElementFocus
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-redesigned.component.html:123]: **_cdkMonitorElementFocus_** is used there as to catch a source of focus event on this element (an indicator step) and style it with outline for Tab key navigation case.

File: ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-redesigned.component.html [hunk 11]
```diff
@@ -0,0 +1,208 @@
...
+                     [attr.tabindex]="isVerticalProgress && (isActive(i) || isCompleted(i)) ? 1 : null"
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-redesigned.component.html:124]: Added **_tabindex_** attribute in order to have a chance to navigate  through the (vertical) indicator steps by keyboard.

File: ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-redesigned.component.html [hunk 12]
```diff
@@ -0,0 +1,208 @@
...
+            (click)="incrementStep()"
+            [innerHTML]="(isPageBusy | async) ? ('SDK.SavingButton' | translate) : ('SDK.NextButton' | translate)">
+    </button>
+</ng-template>
+
+<ng-template #lastStepButtons>
+    <button *ngIf="!isReadonly() && !isAgree()" #submitButton
+            [disabled]="(isPageBusy | async) || dataEntryDisabled"
+            class="button button_medium button_right submit_button"
+            [ngClass]="isVerticalProgress? 'button_secondary' : 'button_primary'"
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-redesigned.component.html:172]: Customized styles for different types of indicators.

**Comment by alextanlan** [comment]
> Would be helpful if you could add a few comments either to code or itself or PR (whichever you think is more appropriate) that can provide a trail of how you approached the changes.

Commented some changes (see above).

**Review by MocanaAtBroad — COMMENTED** [review]
> Reviewed 1 of 13 files at r1.
> *[Reviewable](https://reviewable.io/reviews/broadinstitute/ddp-angular/473)* status: 1 of 11 files reviewed, 17 unresolved discussions (waiting on @alextanlan, @EdPokhvalin, and @fat23cat)

*[a discussion](https://reviewable.io/reviews/broadinstitute/ddp-angular/473#-MVx467lA_VDENt98OwA:-MVx467lA_VDENt98OwB:b-xsa2dd) (no related file):*
Looking good! Just a few comments/questions.

---
___
*[ddp-workspace/projects/ddp-sdk/src/lib/ddp.module.ts, line 251 at r2](https://reviewable.io/reviews/broadinstitute/ddp-angular/473#-MVx0lQVBTEFMIId0q8v:-MVx0lQVBTEFMIId0q8w:b4tqez4) ([raw file](https://github.com/broadinstitute/ddp-angular/blob/4c51c0adcb141d56c7d31805d5e7e9b7c17e9170/ddp-workspace/projects/ddp-sdk/src/lib/ddp.module.ts#L251)):*
> ```XML
>         }),
>         RouterModule,
>         A11yModule
> ```

What is this module doing for us?
___
*[ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/baseActivity.component.ts, line 53 at r2](https://reviewable.io/reviews/broadinstitute/ddp-angular/473#-MVwyF9u1pxw0Ghi4b0E:-MVwyF9v4vBSB7SuBG6e:b-umm5td) ([raw file](https://github.com/broadinstitute/ddp-angular/blob/4c51c0adcb141d56c7d31805d5e7e9b7c17e9170/ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/baseActivity.component.ts#L53)):*
> ```XML
>     public validationRequested = false;
>     public isLoaded = false;
>     public isLoaded$ = new Subject<void>();
> ```

Maybe we can look to consolidate the isLoaded variables? Maybe a boolean BehaviorSubject with init value false? (like isPageBusy below) What do you think. They are doing same thing sort of.
___
*[ddp-workspace/projects/toolkit/src/lib/components/activity/workflow-start-activity-redesigned.component.ts, line 21 at r2](https://reviewable.io/reviews/broadinstitute/ddp-angular/473#-MVwnTi47B6cg0mtE0gr:-MVwnTi47B6cg0mtE0gs:bokyun9) ([raw file](https://github.com/broadinstitute/ddp-angular/blob/4c51c0adcb141d56c7d31805d5e7e9b7c17e9170/ddp-workspace/projects/toolkit/src/lib/components/activity/workflow-start-activity-redesigned.component.ts#L21)):*
> ```XML
>                                [activityGuid]="instanceGuid"
>                                [agreeConsent]="_toolkitConfiguration.agreeConsent"
>                                [activitiesWithVerticalProgress]="_toolkitConfiguration.usesVerticalStepper"
> ```

I think toolkitconfiguration is essentially a global/can be accessed from anywhere, so don't see need to pass it as a input param. I think you can just compute useVerticalStepper from inside ddp-activity-redesigned
___
*[ddp-workspace/projects/toolkit/src/lib/services/toolkitConfiguration.service.ts, line 78 at r2](https://reviewable.io/reviews/broadinstitute/ddp-angular/473#-MVww1-L0VOMj91YIsUs:-MVww1-L0VOMj91YIsUt:b-uuhmm3) ([raw file](https://github.com/broadinstitute/ddp-angular/blob/4c51c0adcb141d56c7d31805d5e7e9b7c17e9170/ddp-workspace/projects/toolkit/src/lib/services/toolkitConfiguration.service.ts#L78)):*
> ```XML
>     dashboardDisplayedColumns: Array<DashboardColumns> = ['name', 'summary', 'date', 'status', 'actions'];
>     // activities which should use a vertical progress indicator
>     usesVerticalStepper = ['FAMILY_HISTORY'];
> ```

I think here value can be empty array. Better to specify FAMILY_HISTORY in the corresponding study (we have been doing it in app.module.ts)

**Review by alextanlan — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/broadinstitute/ddp-angular/473)* status: 1 of 12 files reviewed, 17 unresolved discussions (waiting on @alextanlan, @EdPokhvalin, @fat23cat, and @MocanaAtBroad)

___
*[ddp-workspace/projects/ddp-sdk/src/lib/ddp.module.ts, line 251 at r2](https://reviewable.io/reviews/broadinstitute/ddp-angular/473#-MVx0lQVBTEFMIId0q8v:-MVzrICX6qWGN-34HEo1:b-52dtpc) ([raw file](https://github.com/broadinstitute/ddp-angular/blob/4c51c0adcb141d56c7d31805d5e7e9b7c17e9170/ddp-workspace/projects/ddp-sdk/src/lib/ddp.module.ts#L251)):*
<details><summary><i>Previously, MocanaAtBroad wrote…</i></summary><blockquote>

What is this module doing for us?
</blockquote></details>

It is an Angular module for a11y(accessibility) purposes.
   May be it is over-engineering, but I wanted to have a chance to navigate through the indicator steps by keyboard. 
And I use 'cdkMonitorElementFocus' directive (from A11yMOdule) as to catch a source of focus event on this element (an indicator step) and style it with outline for Tab key navigation case (and not have the outline for mouse click and touch event).
___
*[ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/baseActivity.component.ts, line 53 at r2](https://reviewable.io/reviews/broadinstitute/ddp-angular/473#-MVwyF9u1pxw0Ghi4b0E:-MVzqp4G8__nYh2VLvuL:bbzf5qk) ([raw file](https://github.com/broadinstitute/ddp-angular/blob/4c51c0adcb141d56c7d31805d5e7e9b7c17e9170/ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/baseActivity.component.ts#L53)):*
<details><summary><i>Previously, MocanaAtBroad wrote…</i></summary><blockquote>

Maybe we can look to consolidate the isLoaded variables? Maybe a boolean BehaviorSubject with init value false? (like isPageBusy below) What do you think. They are doing same thing sort of.
</blockquote></details>

I just tried not to break an existing functionality:) I will check what I can consolidate.
___
*[ddp-workspace/projects/toolkit/src/lib/components/activity/workflow-start-activity-redesigned.component.ts, line 21 at r2](https://reviewable.io/reviews/broadinstitute/ddp-angular/473#-MVwnTi47B6cg0mtE0gr:-MVzqD5A7xpauQM3ETWD:b9f0u8) ([raw file](https://github.com/broadinstitute/ddp-angular/blob/4c51c0adcb141d56c7d31805d5e7e9b7c17e9170/ddp-workspace/projects/toolkit/src/lib/components/activity/workflow-start-activity-redesigned.component.ts#L21)):*
<details><summary><i>Previously, MocanaAtBroad wrote…</i></summary><blockquote>

I think toolkitconfiguration is essentially a global/can be accessed from anywhere, so don't see need to pass it as a input param. I think you can just compute useVerticalStepper from inside ddp-activity-redesigned
</blockquote></details>

The thing is that if we inject the toolkit config service (from toolkit module) into activity-redesigned.component (ddp-sdk module), we going to have a circular dependency (ddp-sdk <-> toolkit)
___
*[ddp-workspace/projects/toolkit/src/lib/services/toolkitConfiguration.service.ts, line 78 at r2](https://reviewable.io/reviews/broadinstitute/ddp-angular/473#-MVww1-L0VOMj91YIsUs:-MVzptILCDOrksRjUhTR:b-vflzkn) ([raw file](https://github.com/broadinstitute/ddp-angular/blob/4c51c0adcb141d56c7d31805d5e7e9b7c17e9170/ddp-workspace/projects/toolkit/src/lib/services/toolkitConfiguration.service.ts#L78)):*
<details><summary><i>Previously, MocanaAtBroad wrote…</i></summary><blockquote>

I think here value can be empty array. Better to specify FAMILY_HISTORY in the corresponding study (we have been doing it in app.module.ts)
</blockquote></details>

If we suppose to customize a Family History progress indicator for different activities, I agree. 
Fixed at aa34b6475ad1d87139f0e5989159754a4acdecde

**Review by MocanaAtBroad — CHANGES_REQUESTED** [review]
> Reviewed 1 of 13 files at r1, 1 of 2 files at r3.
> *[Reviewable](https://reviewable.io/reviews/broadinstitute/ddp-angular/473)* status: 3 of 12 files reviewed, 17 unresolved discussions (waiting on @alextanlan, @EdPokhvalin, @fat23cat, and @MocanaAtBroad)

___
*[ddp-workspace/projects/ddp-sdk/src/lib/ddp.module.ts, line 251 at r2](https://reviewable.io/reviews/broadinstitute/ddp-angular/473#-MVx0lQVBTEFMIId0q8v:-MW-l5fy5pp8-adi9fA4:bef2ieo) ([raw file](https://github.com/broadinstitute/ddp-angular/blob/4c51c0adcb141d56c7d31805d5e7e9b7c17e9170/ddp-workspace/projects/ddp-sdk/src/lib/ddp.module.ts#L251)):*
<details><summary><i>Previously, alextanlan (Alex Maratkanov) wrote…</i></summary><blockquote>

It is an Angular module for a11y(accessibility) purposes.
   May be it is over-engineering, but I wanted to have a chance to navigate through the indicator steps by keyboard. 
And I use 'cdkMonitorElementFocus' directive (from A11yMOdule) as to catch a source of focus event on this element (an indicator step) and style it with outline for Tab key navigation case (and not have the outline for mouse click and touch event).
</blockquote></details>

Cool! Had not noticed that. Is it possible to navigate to a specific step using keyboard?
___
*[ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/baseActivity.component.ts, line 53 at r2](https://reviewable.io/reviews/broadinstitute/ddp-angular/473#-MVwyF9u1pxw0Ghi4b0E:-MW-lbpfDrre3twzROCy:bvqucr3) ([raw file](https://github.com/broadinstitute/ddp-angular/blob/4c51c0adcb141d56c7d31805d5e7e9b7c17e9170/ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/baseActivity.component.ts#L53)):*
<details><summary><i>Previously, alextanlan (Alex Maratkanov) wrote…</i></summary><blockquote>

I just tried not to break an existing functionality:) I will check what I can consolidate.
</blockquote></details>

I can sympathize with the risk aversion! I worry that this component has already grown overly complicated. Not your fault! We have continued to pile on functionality. This is one of our best candidates for refactoring in Angular code base. But let's see if we can
___
*[ddp-workspace/projects/toolkit/src/lib/components/activity/workflow-start-activity-redesigned.component.ts, line 21 at r2](https://reviewable.io/reviews/broadinstitute/ddp-angular/473#-MVwnTi47B6cg0mtE0gr:-MW-mpBO8ESotybv3DGY:b-c4mva9) ([raw file](https://github.com/broadinstitute/ddp-angular/blob/4c51c0adcb141d56c7d31805d5e7e9b7c17e9170/ddp-workspace/projects/toolkit/src/lib/components/activity/workflow-start-activity-redesigned.component.ts#L21)):*
<details><summary><i>Previously, alextanlan (Alex Maratkanov) wrote…</i></summary><blockquote>

The thing is that if we inject the toolkit config service (from toolkit module) into activity-redesigned.component (ddp-sdk module), we going to have a circular dependency (ddp-sdk <-> toolkit)
</blockquote></details>

If param value is used in SDK then this is probably indication that config param should live there too.

**Review by alextanlan — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/broadinstitute/ddp-angular/473)* status: 3 of 12 files reviewed, 17 unresolved discussions (waiting on @alextanlan, @EdPokhvalin, @fat23cat, and @MocanaAtBroad)

___
*[ddp-workspace/projects/ddp-sdk/src/lib/ddp.module.ts, line 251 at r2](https://reviewable.io/reviews/broadinstitute/ddp-angular/473#-MVx0lQVBTEFMIId0q8v:-MW0NEYX7o9-341nzro1:by80849) ([raw file](https://github.com/broadinstitute/ddp-angular/blob/4c51c0adcb141d56c7d31805d5e7e9b7c17e9170/ddp-workspace/projects/ddp-sdk/src/lib/ddp.module.ts#L251)):*
<details><summary><i>Previously, MocanaAtBroad wrote…</i></summary><blockquote>

Cool! Had not noticed that. Is it possible to navigate to a specific step using keyboard?
</blockquote></details>

For the vertical indicator  - yes, for now.

**Review by alextanlan — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/broadinstitute/ddp-angular/473)* status: 3 of 12 files reviewed, 17 unresolved discussions (waiting on @alextanlan, @EdPokhvalin, @fat23cat, and @MocanaAtBroad)

___
*[ddp-workspace/projects/toolkit/src/lib/components/activity/workflow-start-activity-redesigned.component.ts, line 21 at r2](https://reviewable.io/reviews/broadinstitute/ddp-angular/473#-MVwnTi47B6cg0mtE0gr:-MW0QDhM2VkIubSE_TBs:b-yq8ttn) ([raw file](https://github.com/broadinstitute/ddp-angular/blob/4c51c0adcb141d56c7d31805d5e7e9b7c17e9170/ddp-workspace/projects/toolkit/src/lib/components/activity/workflow-start-activity-redesigned.component.ts#L21)):*
<details><summary><i>Previously, MocanaAtBroad wrote…</i></summary><blockquote>

If param value is used in SDK then this is probably indication that config param should live there too.
</blockquote></details>

I did not get that point. Let's discuss it tomorrow, shall we?

**Review by MocanaAtBroad — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/broadinstitute/ddp-angular/473)* status: 3 of 12 files reviewed, 17 unresolved discussions (waiting on @alextanlan, @EdPokhvalin, @fat23cat, and @MocanaAtBroad)

___
*[ddp-workspace/projects/toolkit/src/lib/components/activity/workflow-start-activity-redesigned.component.ts, line 21 at r2](https://reviewable.io/reviews/broadinstitute/ddp-angular/473#-MVwnTi47B6cg0mtE0gr:-MW1-K9aDAKh4Y5MitSd:b-tp3gm) ([raw file](https://github.com/broadinstitute/ddp-angular/blob/4c51c0adcb141d56c7d31805d5e7e9b7c17e9170/ddp-workspace/projects/toolkit/src/lib/components/activity/workflow-start-activity-redesigned.component.ts#L21)):*
<details><summary><i>Previously, alextanlan (Alex Maratkanov) wrote…</i></summary><blockquote>

I did not get that point. Let's discuss it tomorrow, shall we?
</blockquote></details>

My point was simply that if this parameter configures something in SDK, then the parameter should probably live in SDK configuration instead of living in toolkit configuration.

**Review by alextanlan — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/broadinstitute/ddp-angular/473)* status: 3 of 9 files reviewed, 17 unresolved discussions (waiting on @alextanlan, @EdPokhvalin, @fat23cat, and @MocanaAtBroad)

___
*[ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/baseActivity.component.ts, line 53 at r2](https://reviewable.io/reviews/broadinstitute/ddp-angular/473#-MVwyF9u1pxw0Ghi4b0E:-MW8sCLz7BvX63GLUHRI:bomn3ca) ([raw file](https://github.com/broadinstitute/ddp-angular/blob/4c51c0adcb141d56c7d31805d5e7e9b7c17e9170/ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/baseActivity.component.ts#L53)):*
<details><summary><i>Previously, MocanaAtBroad wrote…</i></summary><blockquote>

I can sympathize with the risk aversion! I worry that this component has already grown overly complicated. Not your fault! We have continued to pile on functionality. This is one of our best candidates for refactoring in Angular code base. But let's see if we can
</blockquote></details>

I have combined `isLoaded` with `isLoaded$` at ab89881823e6f0ee31f58f55c1fbf84986e1e0c6.
It helped with becoming a bit more logical, but did not help with decreasing code lines :)
___
*[ddp-workspace/projects/toolkit/src/lib/components/activity/workflow-start-activity-redesigned.component.ts, line 21 at r2](https://reviewable.io/reviews/broadinstitute/ddp-angular/473#-MVwnTi47B6cg0mtE0gr:-MW8s3DJ9bYd6Ph-XAa7:b-7q65ot) ([raw file](https://github.com/broadinstitute/ddp-angular/blob/4c51c0adcb141d56c7d31805d5e7e9b7c17e9170/ddp-workspace/projects/toolkit/src/lib/components/activity/workflow-start-activity-redesigned.component.ts#L21)):*
<details><summary><i>Previously, MocanaAtBroad wrote…</i></summary><blockquote>

My point was simply that if this parameter configures something in SDK, then the parameter should probably live in SDK configuration instead of living in toolkit configuration.
</blockquote></details>

Understood. Moved to ddp-sdk config at e4fc8ec35b04dbb3a45a07b88e46fa053a1ff945

**Review by MocanaAtBroad — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/broadinstitute/ddp-angular/473)* status: 3 of 9 files reviewed, 14 unresolved discussions (waiting on @alextanlan, @EdPokhvalin, @fat23cat, and @MocanaAtBroad)

**Pull request merged**